File: Java/2598. Smallest Missing Non-negative Integer After Operations.java
class Solution {
    public int findSmallestInteger(int[] nums, int value) {
        int[] mods = new int[value];
        for (int n: nums) {
            int num = (n % value + value) % value;
            mods[num]++;
        }

        for (int n = 0; n < nums.length; n++) {
            if (mods[n % value] <= 0) {
                return n;
            }
            mods[n % value]--;
        }

        return nums.length;
    }
}

File: Python3/2598. Smallest Missing Non-negative Integer After Operations.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        mods = Counter()
        for n in nums:
            mods[n % value] += 1

        for num in range(len(nums)):
            if mods[num % value] == 0:
                return num
            mods[num % value] -= 1
        
        return len(nums)